src/cli.rs

```diff
@@ -140,6 +140,61 @@ impl GitHubAddOpts {
     }
 }
 
+#[derive(Debug, StructOpt)]
+pub struct ForgejoAddOpts {
+    pub server: String,
+    pub owner: String,
+    pub repository: String,
+
+    #[structopt(flatten)]
+    pub more: GenericGitAddOpts,
+}
+impl ForgejoAddOpts {
+    pub fn add(&self) -> Result<(String, Pin)> {
+        let server_url = match Url::parse(&self.server) {
+            Ok(url) => url,
+            Err(firsterror) => match Url::parse(&("https://".to_string() + self.server.as_str())) {
+                Ok(url) => url,
+                Err(_seconderror) => return Err(firsterror.into()),
+            },
+        };
```

Comment:
```suggestion
        let server_url = Url::parse(&self.server)
            .or_else(|err|
                Url::parse(&("https://".to_string() + self.server.as_str())
                    .map_err(|_| err)
            )
```

maybe something more like this is more concise?